pkg/controller/selector/selector_test.go

```diff
@@ -101,3 +102,732 @@ var (
 		},
 	}
 )
+
+var (
+	PoolA = model.StoragePoolSpec{
+		BaseModel: &model.BaseModel{
+			Id:        "f4486139-78d5-462d-a7b9-fdaf6c797e11",
+			CreatedAt: "2017-10-24T15:04:05",
+		},
+		FreeCapacity:     int64(50),
+		AvailabilityZone: "az1",
+		Extras: model.ExtraSpec{
+			"thin":     true,
+			"dedupe":   true,
+			"compress": true,
+			"diskType": "SSD",
+		},
+	}
+	PoolB = model.StoragePoolSpec{
+		BaseModel: &model.BaseModel{
+			Id:        "f4486139-78d5-462d-a7b9-fdaf6c797e12",
+			CreatedAt: "2017-10-24T15:04:06",
+		},
+		FreeCapacity:     int64(60),
+		AvailabilityZone: "az2",
+		Extras: model.ExtraSpec{
+			"thin":     false,
+			"dedupe":   false,
+			"compress": false,
+			"diskType": "SATA",
+		},
+	}
+	PoolC = model.StoragePoolSpec{
+		BaseModel: &model.BaseModel{
+			Id:        "f4486139-78d5-462d-a7b9-fdaf6c797e13",
+			CreatedAt: "2017-10-24T15:04:07",
+		},
+		FreeCapacity:     int64(70),
+		AvailabilityZone: "az3",
+		Extras: model.ExtraSpec{
+			"thin":     true,
+			"dedupe":   false,
+			"compress": true,
+			"diskType": "SSD",
+		},
+	}
+)
+
+func TestStructToMap(t *testing.T) {
+	poolMap, err := StructToMap(PoolA)
+
+	if nil != err {
+		t.Errorf(err.Error())
+	}
+
+	_, ok := poolMap["freeCapacity"]
+	if !ok {
+		t.Errorf("Expected ok, get %v", ok)
+	}
+
+	_, ok = poolMap["extras.thin"]
+	if !ok {
+		t.Errorf("Expected ok, get %v", ok)
+	}
+}
+
```

Comment:
Please take a look of https://github.com/opensds/opensds/pull/281 and add tests accordingly.

Reply:
This is related to StoragePoolSpec in opensds\pkg\model\pool.go and has nothing to do with  #281